### DISCUSSION TITLE
how does the epdiy handle input?

### DISCUSSION THREAD

**Discussion by SpandexWizard** [vroland/epdiy#334] [General]
so i've been rereading the documentation and examples and I think i've missed something. is there any way to send data to the board from another source? like an hdmi port or the like. or does everything have to be compiled and written to the esp32?

**Comment by martinberlin** [comment]
No. With one of my boards, that I've added a second USB-CDC that goes directly to S3 MCU, you should be able to receive Serial data per USB. Or also add an SD-Card adapter and use it as Mass storage device.
Standard v7 only exposes D+ D- as a header so you should build an adapter to do the same.

![v7 martinberlin](https://cdn.tindiemedia.com/images/resize/JNyabxxPbQhza6AN6isH5kTKDe0=/p/fit-in/994x664/filters:fill(fff)/i/921901/products/2023-10-28T20%3A33%3A11.276Z-8F407914-1FB3-4136-80D1-61660BB8F077..jpeg) 

And of course you have WiFi and BLE if you want to send a picture over the air

**Reply by SpandexWizard** [reply]
mmm. assuming that wifi is off limits, what's the fastest way to transmit data to the device? it sounds like i only really have two options, the explosed d+ d- pins, or the 12C pins?

**Reply by martinberlin** [reply]
Get a board done and try it out. I would say Serial data would be one of the fastest.
If you want hdmi /video input then check Glade project from Wenting Zhang
